common/djangoapps/edxmako/shortcuts.py

```diff
@@ -42,12 +43,20 @@ def marketing_link(name):
         'ENABLE_MKTG_SITE',
         settings.FEATURES.get('ENABLE_MKTG_SITE', False)
     )
+    marketing_urls = configuration_helpers.get_value(
+        'MKTG_URLS',
+        settings.MKTG_URLS
+    )
 
-    if enable_mktg_site and name in settings.MKTG_URLS:
+    if enable_mktg_site and name in marketing_urls:
         # special case for when we only want the root marketing URL
         if name == 'ROOT':
-            return settings.MKTG_URLS.get('ROOT')
-        return settings.MKTG_URLS.get('ROOT') + settings.MKTG_URLS.get(name)
+            return marketing_urls.get('ROOT')
+        # Using urljoin here allows us to enable a marketing site and set
+        # a site ROOT, but still specify absolute URLs for other marketing
+        # URLs in the MKTG_URLS setting
+        # e.g. urljoin('http://marketing.com', 'http://open-edx.org/about') >>> 'http://open-edx.org/about'
+        return urljoin(marketing_urls.get('ROOT'), marketing_urls.get(name))
```

Comment:
I think the pre-existing code assumes you have 'ROOT' configured in your `MKTG_URLS` setting if you have `ENABLE_MKTG_SITE` set to true. I'm okay with that.